test/brain_test.coffee

```diff
@@ -0,0 +1,253 @@
+        # default is 5s
+        @brain.resetSaveInterval 10
+        # make sure autosave is on
+        @brain.setAutoSave true
+
+        @clock.tick(5000)
+        # old interval has passed
+        expect(@brain.save).to.not.have.been.called
+        @clock.tick(5000)
+        # new interval has passed
+        expect(@brain.save).to.have.been.calledOnce
+
+    describe '#close', ->
+      it 'saves', ->
+        sinon.spy @brain, 'save'
+        @brain.close()
+        expect(@brain.save).to.have.been.calledOnce
+
+      it 'emits a close event', ->
+        sinon.spy @brain, 'emit'
+        @brain.close()
+        expect(@brain.emit).to.have.been.calledWith('close')
+
+      it 'saves before emitting the close event', ->
+        sinon.spy(@brain, 'save')
+        sinon.spy(@brain, 'emit').withArgs('close')
+        @brain.close()
+        expect(@brain.save).to.have.been.calledBefore(@brain.emit)
+
+      it 'stops auto-saving', ->
+        # make sure autosave is on
+        @brain.setAutoSave true
+        @brain.close()
+
+        # set up the spy after because 'close' calls 'save'
+        sinon.spy @brain, 'save'
+
+        @clock.tick(2*5000)
+        expect(@brain.save).to.not.have.been.called
+
+    describe '#get', ->
+      it 'returns the saved value', ->
+        @brain.data._private['test-key'] = 'value'
+        expect(@brain.get('test-key')).to.equal('value')
+
+      it 'returns null if object is not found', ->
+        expect(@brain.get('not a real key')).to.be.null
+
+    describe '#set', ->
+      it 'saves the value', ->
+        @brain.set('test-key', 'value')
+        expect(@brain.data._private['test-key']).to.equal('value')
+
+      it 'sets multiple keys at once if an object is provided', ->
+        @brain.data._private =
+          key1: 'val1'
+          key2: 'val1'
+
+        @brain.set
+          key2: 'val2'
+          key3: 'val2'
+
+        expect(@brain.data._private).to.deep.equal
+          key1: 'val1'
+          key2: 'val2'
+          key3: 'val2'
+
+      # Unable to understand why this behavior is needed, but adding a test
+      # case to protect it
+      it 'emits loaded', ->
+        sinon.spy @brain, 'emit'
+        @brain.set('test-key', 'value')
+        expect(@brain.emit).to.have.been.calledWith('loaded', @brain.data)
+
+      it 'returns the brain', ->
+        expect(@brain.set('test-key', 'value')).to.equal(@brain)
+
+    describe '#remove', ->
+      it 'removes the specified key', ->
+        @brain.data._private['test-key'] = 'value'
+        @brain.remove 'test-key'
+        expect(@brain.data._private).to.not.include.keys('test-key')
+
+    describe '#userForId', ->
+      it 'returns the user object', ->
+        expect(@brain.userForId(1)).to.equal(@user1)
+
+      it 'does an exact match', ->
+        user4 = @brain.userForId('FOUR')
+        expect(@brain.userForId('four')).to.not.equal(user4)
+
+      # Cannot understand why this behavior is needed, but adding a test case
```

Comment:
I think this [can be removed](https://github.com/github/hubot/pull/670#issuecomment-74808011), but I figure it is better to have the tests and then remove them later.